### PR TITLE
History and logging

### PR DESCRIPTION
**Pull request by hobinjk-ptc**
Fixes up the existing recorder functionality to add periodic checkpoints and compression. This achieves a compression ratio of 0.2 with just gzip, the disabled compressFloat functionality adds another 0.8

### CODE REVIEW

**Review by dangond-ptc — APPROVED** [review]
Looks good to me!

**Review by benptc — APPROVED** [review]
Looks good! Nice to have these history APIs available, too!

File: libraries/recorder.js [hunk 1]
```diff
@@ -1,70 +1,190 @@
-var fs = require('fs');
+const fs = require('fs');
```
Comment (ptc-mklaudiny) [inline libraries/recorder.js:1]: It would be great for newcomers to have brief comments what main methods do.

File: routers/history.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        res.set('Content-Encoding', 'gzip');
+        const readStream = fs.createReadStream(compressedLogPath);
+        readStream.pipe(res);
+    } else {
+        const readStream = fs.createReadStream(compressedLogPath);
+        readStream.pipe(gzipStream).pipe(res);
+    }
+});
+
+router.get('/patches', function(req, res) {
```
Comment (ptc-mklaudiny) [inline routers/history.js:47]: What kind of data are these patches? RGBD snapshots?

**Review by ptc-mklaudiny — APPROVED** [review]

File: routers/history.js (hunk 2 above)
Comment (hobinjk-ptc) [reply routers/history.js:47]: Yes, this is for persisting the snapshot along with the camera's pose on the server (in addition to the current persistence through window.localStorage)

File: libraries/recorder.js (hunk 1 above)
Comment (hobinjk-ptc) [reply libraries/recorder.js:1]: Good point, thanks!

**Pull request merged**